### ISSUE TITLE
pronto.Ontology() gives inconsistent lengths

### ISSUE THREAD

**Issue by winni-genp** [althonos/pronto#4]
Using the input `test.obo`:
```
[Term]
id: ONT0:CHILD_0
name: child_0
is_a: ONT0:ROOT ! root

[Term]
id: ONT0:CHILD_1
name: child_1
is_a: ONT0:ROOT ! root

[Term]
id: ONT0:CHILD_2
name: child_2
is_a: ONT0:ROOT ! root

[Term]
id: ONT0:ROOT
name: root
```

and running `test.py`:
```
import pronto

print len(pronto.Ontology("./test.obo"))
print len(pronto.Ontology("./test.obo"))
print len(pronto.Ontology("./test.obo"))
print len(pronto.Ontology("./test.obo"))
```

Using `pronto 0.3.3` I get the output:
```
1
2
4
2
```

The expected output as given by `pronto 0.3.2` is:
```
4
4
4
4
```

**Comment by althonos** [comment]
My guess: the parsing is multiprocessed, so getting the length so "early" checks for the length of the Ontology.terms dictionary, which is incomplete at that point.
I think using a Lock will fix the problem.

Thanks for feedback anyway !

**Comment by althonos** [comment]
Actually this seems to be a multiprocessing.Queue problem, but this issue was in `pronto 0.3.2` already, only less frequent.

**Comment by althonos** [comment]
Attempted fix in d95d3594 by changing pronto.utils.Queue behaviour to a JoinableQueue, but this seems to introduce frequent deadlocks, so I won't pull the experimental branch in the master branch right now.
